cpu/atmega2560/periph/gpio.c

```diff
@@ -28,24 +27,232 @@
 
 #if GPIO_NUMOF
 
+/*
 typedef struct {
     void (*cb)(void);
 } gpio_state_t;
+*/
 
-/*static gpio_state_t config[GPIO_NUMOF]; */
-
+/* static gpio_state_t config[GPIO_NUMOF]; */
 
+/* static pin mapping */
+static const uint8_t gpio_pin_map[GPIO_NUMOF] = {
+#if GPIO_0_EN
+    [GPIO_0] = GPIO_0_PIN,
+#endif
+#if GPIO_1_EN
+    [GPIO_1] = GPIO_1_PIN,
+#endif
+#if GPIO_2_EN
+    [GPIO_2] = GPIO_2_PIN,
+#endif
+#if GPIO_3_EN
+    [GPIO_3] = GPIO_3_PIN,
+#endif
+#if GPIO_4_EN
+    [GPIO_4] = GPIO_4_PIN,
+#endif
+#if GPIO_5_EN
+    [GPIO_5] = GPIO_5_PIN,
+#endif
+#if GPIO_6_EN
+    [GPIO_6] = GPIO_6_PIN,
+#endif
+};
 
 int gpio_init_out(gpio_t dev, gpio_pp_t pushpull)
 {
-    return -1;
+    uint8_t pin;
+
+    if (dev > GPIO_NUMOF)
+    {
+        return -1;
+    }
+
+    pin = gpio_pin_map[dev];
+
+    switch (dev)
+    {
+        case GPIO_0:
+            GPIO_0_DDR |= (1 << pin);
+            if (pushpull)
```

Comment:
@ReneHerthel look at the possible [pushpull modes](https://github.com/RIOT-OS/RIOT/blob/master/drivers/include/periph/gpio.h#L142) in RIOT. With your implementation you will set the pull-ups even if the `GPIO_PULLDOWN` is wanted. What I didn't know was that the MCU only supports "no pull-ups" or "pull-ups" and no "pull-downs". Comparin the gpio [API](https://github.com/RIOT-OS/RIOT/blob/master/drivers/include/periph/gpio.h#L167) you can see that you have to `return -1` on an unavailable mode. Compare the implementation for the [arduino-due CPU](https://github.com/RIOT-OS/RIOT/blob/master/cpu/sam3x8e/periph/gpio.c#L498)